### ISSUE TITLE
Enable having octal and hexadecimal escape characters in `toml` config

### ISSUE THREAD

**Issue by ktff** [timberio/vector#2492]
Support having octal and hexadecimal escape characters such as `\x1b`, `\033` in `toml` configuration.

This could be done through adding a feature in the `serde`/`toml` crates.

ref. #2202

**Comment by KannarFr** [comment]
@ktff did you open an issue on toml crate?

**Comment by ktff** [comment]
@KannarFr No, but it should be straight forward to do. Also it's possible `serde` is the crate responsible for parsing the string, or even maybe something further in the dependency chain, so that would be a good thing to determine to avoid a chain of feature issues just linked one to another.

**Comment by KannarFr** [comment]
@binarylogic can you handle this issue? I'm not rust expert, but if required I can take time to learn :D.
